synapse/lib/lmdbslab.py

```diff
@@ -206,55 +206,70 @@ def nameToAbrv(self, name):
     def abrvToName(self, byts):
         return self.abrvToByts(byts).decode()
 
-class HotCount(s_base.Base):
+class HotKeyVal(s_base.Base):
     '''
-    A hot-loop capable counter that only sync's on commit.
+    A hot-loop capable keyval that only syncs on commit.
     '''
     async def __anit__(self, slab, name):
         await s_base.Base.__anit__(self)
 
         self.slab = slab
         self.cache = collections.defaultdict(int)
         self.dirty = set()
+        self.db = self.slab.initdb(name)
+        self.EncFunc, self.DecFunc = self._getEncDecFuncs()
 
-        self.countdb = self.slab.initdb(name)
-
-        for lkey, lval in self.slab.scanByFull(db=self.countdb):
-            self.cache[lkey] = s_common.int64un(lval)
+        for lkey, lval in self.slab.scanByFull(db=self.db):
+            self.cache[lkey] = self.DecFunc(lval)
 
         slab.on('commit', self._onSlabCommit)
 
         self.onfini(self.sync)
 
+    def _getEncDecFuncs(self):
```

Comment:
Any reason this couldnt just simply be a pair of methods that we override?

Reply:
It is just a performance optimization to avoid an extra method call.

Reply:
Fixed via the staticmethod hack.